src/main/java/org/hydev/mcpm/client/database/mirrors/MirrorSelectBoundary.java

```diff
@@ -63,4 +65,26 @@ default List<Pair<Mirror, Integer>> pingMirrors() throws IOException
      * @param mirror A selected mirror
      */
     void setSelectedMirror(Mirror mirror) throws IOException;
+
+    /**
+     * Get the selected mirror as a URI supplier
+     *
+     * @return URI supplier function
+     */
+    default Supplier<URI> selectedMirrorSupplier()
```

Comment:
Don't return a `Callable` here. You're putting too much responsibility on our callee. It's trivial to wrap this method in a lambda at the call site.